google/cloud/bigtable/internal/rowreaderiterator.h

```diff
@@ -54,7 +54,6 @@ class RowReaderIterator : public std::iterator<std::input_iterator_tag, Row> {
 
   Row const& operator*() const& { return *row_; }
   Row& operator*() & { return *row_; }
-  Row const&& operator*() const&& { return *std::move(row_); }
```

Comment:
Do you want to make this a conditional thing, i.e., if GCC && GCC version >= 4.9 or similar? Or is this method really not necessary?

Reply:
> Do you want to make this a conditional thing, i.e., if GCC && GCC version >= 4.9 or similar?

I think that makes more sense, now that you mention it.

> Or is this method really not necessary?

Well, the standard defines it for a reason (it is one of the canonical operators for an iterator), but they happen in very rare circumstances.  None of the code we have uses it for example.